backend/tests/emulator/test_emulator_integration.py

```
@@ -0,0 +1,363 @@
+"""
+import pytest
+import time
+import requests
+
+
+def emulators_running() -> bool:
+    """Check if GCP emulators are running."""
+    try:
+        requests.get("http://127.0.0.1:8080", timeout=1)
+        requests.get("http://127.0.0.1:4443", timeout=1)
+        return True
```

Ruff (0.14.8)

Consider moving this statement to an `else` block

(TRY300)